src/operator/effectivepolicy/groupreconciler.go

```diff
@@ -95,35 +97,59 @@ func (g *GroupReconciler) getAllServiceEffectivePolicies(ctx context.Context) ([
 		if err != nil {
 			return nil, err
 		}
-		if intent, ok := serviceToIntent[service]; ok {
-			ep.ClientIntent = lo.ToPtr(intent)
+		// Ignore intents in deletion process
+		if clientIntents, ok := serviceToIntent[service]; ok && clientIntents.DeletionTimestamp.IsZero() && clientIntents.Spec != nil {
+			ep.Calls = append(ep.Calls, clientIntents.GetCallsList()...)
+			ep.ClientIntentsEventRecorder = injectablerecorder.NewObjectEventRecorder(&g.InjectableRecorder, lo.ToPtr(clientIntents))
 		}
 		epSlice = append(epSlice, ep)
 	}
 
 	return epSlice, nil
 }
 
+// a function that checks if we should create a SEP for a given intent target server
+func (g *GroupReconciler) shouldCreateEffectivePolicyForIntentTargetServer(intent v1alpha3.Intent) bool {
+	if intent.IsTargetOutOfCluster() {
+		return false
+	}
+	if intent.IsTargetServerKubernetesService() {
```

Comment:
```suggestion
        // Services are currently unused when used as a target, since the policy is created by looking at client Calls.
	if intent.IsTargetServerKubernetesService() {
```